plugins/kafka-backend/package.json

```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@backstage/plugin-kafka-backend",
+  "version": "0.1.0",
+  "main": "src/index.ts",
+  "types": "src/index.ts",
+  "license": "Apache-2.0",
+  "private": false,
+  "publishConfig": {
+    "access": "public",
+    "main": "dist/index.cjs.js",
+    "types": "dist/index.d.ts"
+  },
+  "homepage": "https://backstage.io",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/backstage/backstage",
+    "directory": "plugins/kafka-backend"
+  },
+  "keywords": [
+    "backstage",
+    "kafka"
+  ],
+  "configSchema": "config.d.ts",
+  "scripts": {
+    "start": "backstage-cli backend:dev",
+    "build": "backstage-cli backend:build",
+    "lint": "backstage-cli lint",
+    "test": "backstage-cli test",
+    "prepack": "backstage-cli prepack",
+    "postpack": "backstage-cli postpack",
+    "clean": "backstage-cli clean"
+  },
+  "dependencies": {
+    "@backstage/backend-common": "^0.4.1",
+    "@backstage/catalog-model": "^0.6.0",
+    "@backstage/config": "^0.1.2",
+    "@types/express": "^4.17.6",
+    "express": "^4.17.1",
+    "express-promise-router": "^3.0.3",
+    "kafkajs": "^1.16.0-beta.6",
+    "lodash": "^4.17.15",
+    "winston": "^3.2.1"
+  },
+  "devDependencies": {
+    "@backstage/cli": "^0.4.3",
+    "@types/jest-when": "^2.7.2",
+    "@types/lodash": "^4.14.151",
+    "jest-extended": "^0.11.5",
+    "jest-when": "^3.1.0",
```

Comment:
Would prefer not to add these two jest extensions tbh. They pull in quite a lot of extra deps and I'm not sure about their value. Thoughts @backstage/maintainers ?

Reply:
hmm... I think I can work around not using `jest-extended` by asserting on `Set(list)` instead of the `list` itself, but `jest-when` is super useful for creating Mockito-like mocks which I need [here](https://github.com/backstage/backstage/pull/3985/files#diff-22871568ab364826be1893bf3f65e59f50afe5caa194d3cacdf46c028d59efab) and it's also recommended by [ThoughtWorks](https://www.thoughtworks.com/radar/languages-and-frameworks/jest-when#:~:text=Trial%3F-,jest%2Dwhen%20is%20a%20lightweight%20JavaScript%20library%20that%20complements%20Jest,of%20modules%20with%20many%20dependencies.)

Reply:
Yeah I agree `jest-when` looks pretty good. Let's keep it but make sure we're bringing in the minimum amount of extra dependencies in the lockfile

Reply:
Good question. Do you have further concerns besides keeping the yarn install part of builds down?

Reply:
Familiarity when reading tests just to avoid having to bounce to docs. Can't say I obviously knew what `toIncludeSameMembers` did tbh. I think `jest-when` is a lot clearer though, more obvious how that works.
 There's also the overhead of having more dependencies and more code in `setupTests.ts`

Not a super high bar, but tbh don't think `jest-extended` looks too tempting either.

In this case I'm also worried about types though, since we're extending the `expect` types that'll be applied to the entire project afaik, which is pretty annoying and confusing in places where we don't install `jest-extended`. Basically if we think `jest-extended` looks good I think it should be added everywhere and possibly even to the CLI, which is a much higher bar. But I don't think it looks useful enough so thinking we shouldn't

Reply:
Sure, removed dependency on `jest-extended`

Reply:
Agreed, having looked at the docs it sure looks like an ok decision to keep jest-when out of the two.